embed/etcd.go

```diff
@@ -318,16 +318,25 @@ func (e *Etcd) serve() (err error) {
 		}(l)
 	}
 
-	// Start a client server goroutine for each listen address
-	ch := http.Handler(&cors.CORSHandler{
-		Handler: v2http.NewClientHandler(e.Server, e.Server.Cfg.ReqTimeout()),
-		Info:    e.cfg.CorsInfo,
-	})
+	v2h := func() http.Handler {
```

Comment:
`v2h` is nil if `e.Config().EnableV2 == false` , the function returns nil and sets `v2h` to nil.

but i agree it unnecessarily complicated, will change to the way you suggested.